### DISCUSSION TITLE
Error while updating LXC zigbee2mqtt container

### DISCUSSION THREAD

**Discussion by mguedes71** [tteck/Proxmox#1014] [Help Needed] [answered]
I'm trying to update the zigbee2mqtt container, but it's giving me the error:

✓ No Backup Directory Exists
 ✓ Stopped Zigbee2MQTT
 ✓ Created Backup of Configuration
 - Updating Zigbee2MQTT...
[ERROR] in line 356: exit code 1: while executing command git pull &> /dev/null

Can someone help me?

**Comment by tteck** [comment, marked as answer]
https://github.com/tteck/Proxmox/issues/1009

**Reply by tteck** [reply]
By the way, the improved error handling eliminated the need for me to search for line 356.

**Comment by tteck** [comment]
I am incorporating this code into the update script.
```
echo "Initiating update"
if ! git pull; then
  echo "Update failed, temporarily storing changes and trying again."
  git stash && git pull || (echo "Update failed even after storing changes. Aborting."; exit 1)
fi
```
This should simplify the process.
